octoprint_mrbeam/software_update_information.py

```diff
@@ -479,35 +481,36 @@ def _get_curent_version(input_moduleconfig, module_id, plugin):
     return current_version
 
 
-def _generate_config_of_beamos(moduleconfig, beamos_date, tierversion):
+def _generate_config_of_beamos(moduleconfig, beamos_version, tierversion):
     """
-    generates the config for the given beamos_date of the tierversion
+    generates the config for the given beamos_version of the tierversion
     Args:
         moduleconfig: update config of the module
-        beamos_date: date of the beamos
+        beamos_version: version of the beamos
         tierversion: software tier
 
     Returns:
         beamos config of the tierversion
     """
-    if "beamos_date" not in moduleconfig:
+    if "beamos_version" not in moduleconfig:
         return {}
 
-    beamos_date_config = {}
-    prev_beamos_date_entry = datetime.strptime("2000-01-01", "%Y-%m-%d").date()
-    for date, beamos_config in moduleconfig["beamos_date"].items():
+    beamos_version_config = {}
+    # TODO change to version instead of date
+    prev_beamos_version_entry = "0.0.0"
+    for beamos_version, beamos_config in moduleconfig["beamos_version"].items():
         if (
-            beamos_date >= datetime.strptime(date, "%Y-%m-%d").date()
-            and prev_beamos_date_entry < beamos_date
+            version.parse(beamos_version) >= version.parse(beamos_version)
```

Comment:
this is the same variable in both sides right ?

Reply:
You are right I have to check